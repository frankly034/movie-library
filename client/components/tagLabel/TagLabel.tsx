"use client";

import styled from "styled-components";

const DisplayTag = styled.span`
  border-radius: ${({ theme }) => theme.variables.borderRadius};
  font-size: 12px;
  padding: 2px 4px;
  background-color: ${({ theme }) => theme.colors.mediumGray};
  color: ${({ theme }) => theme.palette.foregroundSecondary};
`;

export default DisplayTag;
