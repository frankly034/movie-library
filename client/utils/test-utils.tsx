import React, { ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";

import { ThemeProvider } from "../styles";

const Providers = ({ children }: { children: React.ReactNode }) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: Providers, ...options });

export * from "@testing-library/react";
export { customRender as render };
