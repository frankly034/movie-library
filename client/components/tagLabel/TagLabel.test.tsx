import { render } from "../../utils/test-utils";

import TagLabel from ".";

describe("Component - TagLabel", () => {
  it("should renders the component", () => {
    const { container } = render(<TagLabel>Horror</TagLabel>);
    expect(container).toMatchSnapshot();
  });

  it("should render text", () => {
    const { getByText } = render(<TagLabel>Documentary</TagLabel>);
    expect(getByText("Documentary")).toBeTruthy;
  });
});
