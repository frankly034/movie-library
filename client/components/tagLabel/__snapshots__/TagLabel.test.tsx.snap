// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - TagLabel should renders the component 1`] = `
<div>
  <span
    class="sc-bcPKhP Dillv"
  >
    Horror
  </span>
</div>
`;
